### ISSUE TITLE
Infinite build time of CLI tool on Julia 1.10-beta1

### ISSUE THREAD

**Issue by MilesCranmer** [JuliaLang/julia#50839]
x-post from https://github.com/comonicon/Comonicon.jl/issues/249 as it may be a Julia bug.

I am experiencing infinite build times when compiling a Comonicon.jl-enabled CLI tool ([AirspeedVelocity.jl](https://github.com/MilesCranmer/AirspeedVelocity.jl)) on Julia 1.10-beta1. Earlier versions of Julia such as 1.9.2 work fine.

**To Reproduce**

```bash
julia +1.10 -e 'using Pkg; Pkg.add("AirspeedVelocity")'
```
and, if needed, `Pkg.build("AirspeedVelocity")`.

**Screenshots**
<img width="921" alt="Screenshot 2023-08-08 at 10 21 43" src="https://github.com/comonicon/Comonicon.jl/assets/7593028/9a9a4d7e-2478-49e9-b49e-709886e95cff">

^ It gets stuck here

The log it mentions contains the following:

```
[pid 61578] waiting for IO to finish:
 TYPE[FD/PID]       @UV_HANDLE_T->DATA
 pipe[16]           @0x112e2e790->0x104604bf0
 pipe[10]           @0x112e92600->0x104604c90
 process[61589]     @0x112e93dd0->0x11423eaf0
```

which is repeated over and over again.

More info:

```
julia> versioninfo()
Julia Version 1.10.0-beta1
Commit 6616549950e (2023-07-25 17:43 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin22.4.0)
  CPU: 8 × Apple M1 Pro
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-15.0.7 (ORCJIT, apple-m1)
  Threads: 8 on 6 virtual cores
Environment:
  JULIA_NUM_THREADS = auto
  JULIA_FORMATTER_SO = /Users/mcranmer/julia_formatter.so
  JULIA_EDITOR = code
```

cc @Roger-luo

**Comment by vtjnash** [comment]
It is waiting for pid=61589 to exit, which is a package bug.

**Comment by MilesCranmer** [comment]
It’s weird it only occurs on 1.10. Is there a way to get more debugging info out of the build step?

**Comment by vtjnash** [comment]
Older versions accidentally aborted it instead, which was worse for many other reasons.

**Comment by MilesCranmer** [comment]
cc @timholy 

Update: I found the issue by doing a binary search of commenting out code. It turns out that it was the fault of one of my dependencies, PlotlyKaleido.jl. Apparently this package has some I/O resource hanging around, making precompilation/building hang.

Somehow that package managed to precompile on its own, but using it as a dependency made precompilation of my own package hang.

The following code gives the exact same issue I was seeing:

```julia
module AirspeedVelocity

using PlotlyKaleido

end
```

Is there a way to localize this I/O issue in the released Julia v1.10 so it is easier to track down?

**Comment by MilesCranmer** [comment]
I fixed it in the dependency with this: https://github.com/JuliaPlots/PlotlyKaleido.jl/pull/6

However there was really no easy way of finding this besides randomly turning on/off parts of code. In case others stumble upon this thread, what's the best way of tracking something like this down? Or could such a thing be added to v1.10?
